pkgs/development/haskell-modules/configuration-nix.nix

```diff
@@ -750,4 +750,9 @@ self: super: builtins.intersectAttrs super {
     '';
   });
 
+  # Fix compilation of Setup.hs and disable tests since it's empty
+  guid = overrideCabal (super.guid) (drv: {
+    prePatch = "sed -i '1d' Setup.hs";
```

Comment:
What is this doing?

Also, is this an upstream problem?  If so, please report it upstream.

Reply:
This removes the module declaration at the first line of Setup.hs.

Upstream already notified about it but unresponsive. This package is dependency of many packages in haskell ecosystem and I'm trying to fix it to use nix in unison compiler source.

Reply:
> This removes the module declaration at the first line of Setup.hs

Could you add a comment about this right above the line 755?  It should be obvious to anyone even if they haven't personally looked at the problematic `Setup.hs` file.

Also, could you link to the upstream issue?  This is so that it is easy for us to know when a new (fixed) version is released, so we can remove this override.

--------------------------------------

Two other things:

- Could you rebase this on the `haskell-updates` branch and change the base here on github to `haskell-updates`?  We generally send haskell fixes to the `haskell-updates` branch (not `master`).

- Could you remove `guid` from the list of `broken-packages` in `pkgs/development/haskell-modules/configuration-hackage2nix.yaml`?  If you don't do this, then `broken = true` will always be set in the `pkgs/development/haskell-modules/hackage-packages.nix` file.